### PR TITLE
Update SonarLint config

### PR DESCRIPTION
**Pull request by martin-strecker-sonarsource**

### CODE REVIEW

File: analyzers/.sonarlint/sonaranalyzer-dotnetCSharp.ruleset [hunk 1]
```diff
@@ -298,7 +298,7 @@
     <Rule Id="S4015" Action="Warning" />
     <Rule Id="S4016" Action="None" />
     <Rule Id="S4017" Action="Warning" />
-    <Rule Id="S4018" Action="Info" />
+    <Rule Id="S4018" Action="None" />
```
Comment (martin-strecker-sonarsource) [inline analyzers/.sonarlint/sonaranalyzer-dotnetCSharp.ruleset:301]: S4018 - All type parameters should be used in the parameter list to enable type inference

File: analyzers/.sonarlint/sonaranalyzer-dotnetCSharp.ruleset [hunk 2]
```diff
@@ -332,7 +332,7 @@
     <Rule Id="S4210" Action="Warning" />
     <Rule Id="S4211" Action="Warning" />
     <Rule Id="S4212" Action="Warning" />
-    <Rule Id="S4214" Action="Warning" />
+    <Rule Id="S4214" Action="None" />
```
Comment (martin-strecker-sonarsource) [inline analyzers/.sonarlint/sonaranalyzer-dotnetCSharp.ruleset:335]: S4214 - "P/Invoke" methods should not be visible

File: analyzers/.sonarlint/sonaranalyzer-dotnetCSharp.ruleset [hunk 3]
```diff
@@ -345,7 +345,7 @@
     <Rule Id="S4428" Action="Warning" />
     <Rule Id="S4433" Action="Warning" />
     <Rule Id="S4456" Action="Warning" />
-    <Rule Id="S4457" Action="Warning" />
+    <Rule Id="S4457" Action="None" />
```
Comment (martin-strecker-sonarsource) [inline analyzers/.sonarlint/sonaranalyzer-dotnetCSharp.ruleset:348]: S4457 - Parameter validation in "async"/"await" methods should be wrapped

**Review by costin-zaharia-sonarsource — APPROVED** [review]
LGTM!

**Pull request merged**